tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+    "sof-ipc4/adl-n/sof-adl-n.ri",
+    "sof-ipc4/adl-n/community/sof-adl-n.ri",
+    "sof-ipc4/adl-n/intel-signed/sof-adl-n.ri",
+    "sof-ipc4/tgl-h/community/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/intel-signed/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/sof-tgl-h.ri",
+    "sof-ipc4/mtl/community/sof-mtl.ri",
+    "sof-ipc4/mtl/intel-signed/sof-mtl.ri",
+    "sof-ipc4/mtl/sof-mtl.ri",
+    "sof-ipc4/rpl-s/community/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/intel-signed/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/sof-rpl-s.ri",
+    "sof-ipc4/adl-s/community/sof-adl-s.ri",
+    "sof-ipc4/adl-s/intel-signed/sof-adl-s.ri",
+    "sof-ipc4/adl-s/sof-adl-s.ri",
+    "sof-ipc4/tgl/community/sof-tgl.ri",
+    "sof-ipc4/tgl/intel-signed/sof-tgl.ri",
+    "sof-ipc4/tgl/sof-tgl.ri",
+    "sof-ipc4/adl/sof-adl.ri",
+    "sof-ipc4/adl/community/sof-adl.ri",
+    "sof-ipc4/adl/intel-signed/sof-adl.ri",
+ ]
+
+BASE="/lib/firmware/intel"
+
+def check_fw_files_kernel_to_bundle():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        if os.path.isfile(full_path):
+            if verbose_mode: print("File %s found." % full_path);
+        else:
+            raise Exception("File %s NOT found." % full_path);
+
+def check_fw_files_bundle_to_kernel():
+    #for entry in os.scandir(bundle_base):
+    paths = pathlib.Path(bundle_base).glob("sof*/*/*.ri")
+    for path in paths:
+        bundle_path  = path.relative_to(bundle_base)
+        if not str(bundle_path) in fw_list:
+            raise Exception("Firmware %s not expected by SOF Linux driver" % str(bundle_path))
+
+def check_sign_key():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        try:
+            fw_bin = sof_ri_info.parse_fw_bin(full_path, False, False)
+        except Exception as e:
+            print("FW parse error for %s: %s" % (full_path, str(e)))
+            pass
+
+        if fw_bin == None:
+            return
+
+        hdr = sof_ri_info.getCssManifest(fw_bin).cdir['css_mft_hdr']
+        for attr in hdr.attribs:
+            if attr.name == "modulus":
+                if "/community" in i:
+                    if attr.val != sof_ri_info.COMMUNITY_KEY and \
+                       attr.val != sof_ri_info.COMMUNITY_KEY2:
+                        raise Exception("FW %s not signed with community key" % full_path)
+
+                elif "/intel-signed" in i:
+                    if attr.val == sof_ri_info.COMMUNITY_KEY or \
+                       attr.val == sof_ri_info.COMMUNITY_KEY2 or \
+                       not attr.val in sof_ri_info.KNOWN_KEYS:
+                        raise Exception("Intel signed firmware signed with unexpected key")
+
+
+def check_ipc_version():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        try:
+            fw_bin = sof_ri_info.parse_fw_bin(full_path, False, False)
+        except Exception as e:
+            print("FW parse error for %s: %s" % (full_path, str(e)))
+            pass
```

Comment:
How is a such a parse error acceptable?

Reply:
We have a set of old binaries for Baytrail and friends which fail above. I haven't had time to debug this yet -- probable simple explanation is these are not signed, so the CSE manifest parsing won't work.

```
FW parse error for sof-bin-2024.03//sof/sof-cht.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
FW parse error for sof-bin-2024.03//sof/sof-byt.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
FW parse error for sof-bin-2024.03//sof/sof-bdw.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
FW parse error for sof-bin-2024.03//sof/sof-cht.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
FW parse error for sof-bin-2024.03//sof/sof-byt.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
FW parse error for sof-bin-2024.03//sof/sof-bdw.ri: ('CSE Manifest magic number NOT found, instead: (Reef)', -4)
```

Reply:
Yes these are simply not signed at all. Unfortunately, the current code ignores just not unsigned files but ALL parsing errors. Not OK.

BTW https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html

Reply:
Either fix the parser to detect `(Reef)` and return something useful, or hardcode a short list of "unsigned" platforms here (which is never going to grow now) and return IPC3.